programs/mango-v4/src/instructions/perp_edit_market.rs

```diff
@@ -74,112 +70,128 @@ pub fn perp_edit_market(
+        require_group_admin = true;
     }
     if let Some(max_funding) = max_funding_opt {
         perp_market.max_funding = I80F48::from_num(max_funding);
+        require_group_admin = true;
     }
     if let Some(impact_quantity) = impact_quantity_opt {
         perp_market.impact_quantity = impact_quantity;
+        require_group_admin = true;
     }
     if let Some(fee_penalty) = fee_penalty_opt {
         perp_market.fee_penalty = fee_penalty;
+        require_group_admin = true;
     }
 
-    // unchanged -
-    // long_funding
-    // short_funding
-    // funding_last_updated
-    // open_interest
-    // seq_num
-    // fees_accrued
-    // bump
-
     if let Some(base_decimals) = base_decimals_opt {
         perp_market.base_decimals = base_decimals;
+        require_group_admin = true;
     }
 
-    // unchanged -
-    // perp_market_index
-
-    // unchanged -
-    // registration_time
-
     if let Some(group_insurance_fund) = group_insurance_fund_opt {
         perp_market.set_elligible_for_group_insurance_fund(group_insurance_fund);
+        require_group_admin = true;
     }
 
     if let Some(settle_fee_flat) = settle_fee_flat_opt {
         perp_market.settle_fee_flat = settle_fee_flat;
+        require_group_admin = true;
     }
     if let Some(settle_fee_amount_threshold) = settle_fee_amount_threshold_opt {
         perp_market.settle_fee_amount_threshold = settle_fee_amount_threshold;
+        require_group_admin = true;
     }
     if let Some(settle_fee_fraction_low_health) = settle_fee_fraction_low_health_opt {
         perp_market.settle_fee_fraction_low_health = settle_fee_fraction_low_health;
+        require_group_admin = true;
     }
 
     if let Some(stable_price_delay_interval_seconds) = stable_price_delay_interval_seconds_opt {
         // Updating this makes the old delay values slightly inconsistent
         perp_market.stable_price_model.delay_interval_seconds = stable_price_delay_interval_seconds;
+        require_group_admin = true;
     }
     if let Some(stable_price_delay_growth_limit) = stable_price_delay_growth_limit_opt {
         perp_market.stable_price_model.delay_growth_limit = stable_price_delay_growth_limit;
+        require_group_admin = true;
     }
     if let Some(stable_price_growth_limit) = stable_price_growth_limit_opt {
         perp_market.stable_price_model.stable_growth_limit = stable_price_growth_limit;
+        require_group_admin = true;
     }
 
     if let Some(settle_pnl_limit_factor_opt) = settle_pnl_limit_factor_opt {
         perp_market.settle_pnl_limit_factor = settle_pnl_limit_factor_opt;
+        require_group_admin = true;
     }
     if let Some(settle_pnl_limit_window_size_ts) = settle_pnl_limit_window_size_ts_opt {
         perp_market.settle_pnl_limit_window_size_ts = settle_pnl_limit_window_size_ts;
+        require_group_admin = true;
     }
 
     if let Some(reduce_only) = reduce_only_opt {
         perp_market.reduce_only = u8::from(reduce_only);
     };
```

Comment:
We should only allow the security authority to switch on reduce only, not to switch it off.